src/UserAccountDataChecker.php

```diff
@@ -24,13 +24,13 @@ class UserAccountDataChecker extends DataChecker {
 	public $mUser = null;
 
 	public function __construct() {
-		$this->addCheck( array( &$this, 'checkDomainValidity' ), array() );
-		$this->addCheck( array( &$this, 'checkDomainUser' ), array() );
-		$this->addCheck( array( &$this, 'checkCreatePermissions' ), array() );
-		$this->addCheck( array( &$this, 'checkSorbs' ), array() );
-		$this->addCheck( array( &$this, 'checkUserExists' ), array() );
-		$this->addCheck( array( &$this, 'checkPasswordLength' ), array() );
-		$this->addCheck( array( &$this, 'checkEmailValidity' ), array() );
+		$this->addCheck( array( $this, 'checkDomainValidity' ), array() );
```

Comment:
For 
```
global $wgHooks;
$wgHooks['BeforePageDisplay'][] = array( &$this, 'BeforePageDisplay' );
```

where `&$this` holds the pass-by-reference (mind the `&`) to an instance then the equivalent would be:

```
global $wgHooks;
$that = $this;
$wgHooks['BeforePageDisplay'][] = array( &$that, 'BeforePageDisplay' );
```

Replacing `$this` with `$that` because the line `$that = $this;` copies the reference (all this voodoo is needed because since PHP 7.1 `&$this` can no longer be used in a callback).